model/textparse/interface.go

```diff
@@ -61,12 +61,23 @@ type Parser interface {
 }
 
 // New returns a new parser of the byte slice.
-func New(b []byte, contentType string) Parser {
+//
+// If content type is empty (NewPromParser(b), nil) is returned.
+// Any other invalid content type will return (nil, err).
+func New(b []byte, contentType string) (Parser, error) {
+	if contentType == "" {
+		return NewPromParser(b), nil
+	}
+
 	mediaType, _, err := mime.ParseMediaType(contentType)
-	if err == nil && mediaType == "application/openmetrics-text" {
-		return NewOpenMetricsParser(b)
+	if err != nil {
+		return nil, err
```

Comment:
This is a breaking change, it would not be acceptable.

Reply:
fixed, thanks